torch/csrc/jit/codegen/cuda/executor_utils.cpp

```diff
@@ -777,7 +777,12 @@ kir::ExpressionEvaluator bindKernelInputs(
           "Something went wrong configuring launch. Inputs no longer match.");
 
       for (const auto dim : c10::irange(root_domain.size())) {
-        const auto extent = root_domain[dim]->extent();
+        Val* extent = nullptr;
+        if (root_domain[dim]->hasExpandedExtent()) {
+          extent = root_domain[dim]->expandedExtent();
+        } else {
+          extent = root_domain[dim]->extent();
+        }
```

Comment:
I wonder if this and the one below are the only places we need to change the code as done in this PR. If an IterDomain has an expanded extent, shouldn't it always return the expanded extent?

Reply:
Another change: https://github.com/csarofeen/pytorch/pull/1863/files#diff-296b2445f5a818c3240b26185b37b309fa554f56b909f73109917cbdfd580333R1191-R1195

Reply:
We don't want to always expose expanded extent because we don't want to generate allocations or loop nests based on it.

Reply:
An expanded extent is a non-physical extent on a broadcast domain until the broadcast is realized, then it becomes a physical extent. I don't think we should be binding inputs based on expanded extents.

Reply:
If an expanded extent is an output then that domain should be changed in the tensor metadata, not the allocation.

Reply:
> An expanded extent is a non-physical extent on a broadcast domain until the broadcast is realized

If so, what about merging an expanded broadcast domain with a non-broadcast domain? I'm not really sure what the right extent should be. In Jie's #1772, the merged domain has the extent computed with the expanded extent.  

```
  } else if (outer->isBroadcast() || inner->isBroadcast()) {
    itype = IterType::Iteration;
    if (outer->isBroadcast() && outer->hasExpandedExtent()) {
      merged_id_size = mul(outer->expandedExtent(), inner->extent());
    } else if (inner->isBroadcast() && inner->hasExpandedExtent()) {
      merged_id_size = mul(outer->extent(), inner->expandedExtent());
    }
  }
```
https://github.com/csarofeen/pytorch/pull/1863/files#diff-296b2445f5a818c3240b26185b37b309fa554f56b909f73109917cbdfd580333R1189-R1196

Reply:
Yeah, I'm trying to think it through. I think we might need some better rules for expand. I think with merge we should still treat it like we would with broadcast merged in, but I think it would also need an expanded extent. I just don't know how it would impact anything unless we support view operations.

With a view operation I think it would have to concretize this domain.

Reply:
Feeling like trivial reduction all over again

Reply:
If we need a rule for merge, I would think we need a rule for split too.